### PR TITLE
OpenThread rx failed implemented

### PR DESCRIPTION
**Pull request by MarekPorwisz**
Rx failed event is used by thread for gathering statistical data.

### CODE REVIEW

File: subsys/net/lib/openthread/platform/radio.c [hunk 1]
```diff
@@ -50,6 +50,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_OPENTHREAD_L2_LOG_LEVEL);
 enum pending_events {
 	PENDING_EVENT_FRAME_TO_SEND, /* There is a tx frame to send  */
 	PENDING_EVENT_FRAME_RECEIVED, /* Radio has received new frame */
+	PENDING_EVENT_RX_FAILED, /* The RX failed - this is usually normal */
```
Comment (rlubos) [inline subsys/net/lib/openthread/platform/radio.c:53]: Not sure if I understand this comment: `this is usually normal` - normal what?

File: include/net/ieee802154_radio.h [hunk 2]
```diff
@@ -48,7 +48,15 @@ enum ieee802154_filter_type {
...
-	IEEE802154_EVENT_TX_STARTED /* Data transmission started */
+	IEEE802154_EVENT_TX_STARTED, /* Data transmission started */
+	IEEE802154_EVENT_RX_FAILED   /* Data reception failed */
+};
+
+enum ieee802154_rx_fail_reason {
+	IEEE802154_RX_FAIL_NOT_RECEIVED,  /* Nothing received */
+	IEEE802154_RX_FAIL_INVALID_FSC,   /* Frame had invalid checksum */
+	IEEE802154_RX_FAIL_ADDR_FILTERED, /* Address did not match */
+	IEEE802154_RX_FAIL_FAILED         /* general reason */
```
Comment (rlubos) [inline include/net/ieee802154_radio.h:59]: Not sure about this name, `FAILED` would be true to any other failure as well, how about `IEEE802154_RX_FAIL_OTHER`/`IEEE802154_RX_FAIL_UNKNOWN`?

File: drivers/ieee802154/ieee802154_nrf5.c [hunk 3]
```diff
@@ -630,7 +630,33 @@ void nrf_802154_received_timestamp_raw(uint8_t *data, int8_t power, uint8_t lqi,
...
+	enum ieee802154_rx_fail_reason reason;
+
+	switch (error) {
+	case NRF_802154_RX_ERROR_INVALID_FRAME:
+	case NRF_802154_RX_ERROR_DELAYED_TIMEOUT:
+		reason = IEEE802154_RX_FAIL_NOT_RECEIVED;
+		break;
+
+	case NRF_802154_RX_ERROR_INVALID_FCS:
+		reason = IEEE802154_RX_FAIL_INVALID_FSC;
```
Comment (aurel32) [inline drivers/ieee802154/ieee802154_nrf5.c:642]: You probably want to use `IEEE802154_RX_FAIL_INVALID_FCS` here instead of `INVALID_FSC`.

**Comment by MarekPorwisz** [comment]
I don't think I did any change related to the failure:
START - test_sem_take_timeout

    Assertion failed at WEST_TOPDIR/zephyr/tests/kernel/mem_protect/sys_sem/src/main.c:277: test_sem_take_timeout: (ret_value == 0 is false)
sys_sem_take failed when its shouldn't have
 FAIL - test_sem_take_timeout

How can I restart the test?

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**